src/inputs/OptionSelect.tsx

```diff
@@ -1,76 +1,16 @@
 import React, { Component } from 'react';
-import { computed } from 'mobx';
-import { inject, observer } from 'mobx-react';
-import { isBoolean } from 'lodash';
+import { observer } from 'mobx-react';
 import autoBindMethods from 'class-autobind-decorator';
 
-import * as Antd from 'antd';
+import { IInputProps } from '../interfaces';
 
-import {
-  IAntFormField,
-  IFieldConfig,
-  IFieldConfigOptionSelect,
-  IInjected,
-  IInputProps,
-  IOption,
-} from '../interfaces';
+import ObjectSelect from './ObjectSelect';
 
-import { getOptions } from '../utilities';
-
-export interface IOptionSelectProps {
-  fieldConfig: IFieldConfig;
-  renderOption?: (option: IOption) => React.ReactNode;
-}
-
-// 8 is the most number of options you can show with no scroll
-export const SHOW_OPTION_SEARCH_IF_OVER = 8;
-
-@inject('getOptions')
 @autoBindMethods
 @observer
-class OptionSelect extends Component<IOptionSelectProps> {
-  private get injected () {
-    return this.props as IOptionSelectProps & IInjected & IInputProps & IAntFormField;
-  }
-
-  private get fieldConfig () {
-    return this.props.fieldConfig as IFieldConfigOptionSelect;
-  }
-
-  @computed
-  private get options (): IOption[] {
-    return getOptions(this.fieldConfig, this.injected);
-  }
-
-  private get showSearch (): boolean {
-    // the showSearch fieldConfig option will override this
-    if (isBoolean(this.fieldConfig.showSearch)) {
-      return this.fieldConfig.showSearch;
-    }
-
-    return this.options.length > SHOW_OPTION_SEARCH_IF_OVER;
-  }
-
+class OptionSelect extends Component<IInputProps> {
```

Comment:
Since you set this to receive `IInputProps`, won't this result in a type error if you pass anything in `IObjectSelectProps` to this component?